subsys/spm/secure_services.c

```diff
@@ -3,11 +3,40 @@
  *
  * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
  */
+#include <errno.h>
 #include <cortex_m/tz.h>
 #include <misc/reboot.h>
 #include <autoconf.h>
+#include <secure_services.h>
+
+#ifdef CONFIG_SPM_SERVICE_RNG
+#ifdef MBEDTLS_CONFIG_FILE
+#include MBEDTLS_CONFIG_FILE
+#else
+#include "mbedtls/config.h"
+#endif /* MBEDTLS_CONFIG_FILE */
+
+#include <mbedtls/platform.h>
+#include <mbedtls/entropy_poll.h>
+
+/** @brief Structure holding the memory required to generate entropy
+ */
+static mbedtls_rng_workbuf_internal rng_workbuf;
+#endif
+
+int spm_secure_services_init(void)
+{
+	int err = 0;
+
+#ifdef CONFIG_SPM_SERVICE_RNG
+	static mbedtls_platform_context platform_ctx = {
+		.p_rnd_workbuf = &rng_workbuf
+	};
+	err = mbedtls_platform_setup(&platform_ctx);
```

Comment:
About the use of 'static', is it intentional to place 'platform_ctx' on the heap instead of on the stack?

Reply:
Do you mean 'data segment' instead of 'heap'?

Reply:
right, sorry

Reply:
It is intentional to put it in data segment. Not doing so will require using mbedtls_calloc for a 6 KB buffer of data at unknown time. Acquiring the same amount of stack during calls to init/generate RNG is also problematinc for the same reason

Reply:
I can understand that rng_workbuf can't be on the stack, but I don't see how placing platform_ctx on the stack will require us to use mbedtls_calloc.

Reply:
Ah.. yes. That is true. The static can be removed for the mbedtls_platform_context, yes

Reply:
Done